avalon/nuke/pipeline.py

```diff
@@ -132,31 +140,48 @@ def update_container(node, keys=dict()):
         node (object): nuke node with updated container data
     """
 
-    container = parse_container(node)
+    data = lib.get_avalon_knob_data(node)
+
+    if not isinstance(data, dict):
+        return
+
+    # If not all required data return the empty container
+    required = ['schema', 'id', 'name',
+                'namespace', 'objectName', 'representation', 'version']
+    container = dict()
+
+    # check if key in data
+    for k in required:
+        if not data.get(k, 0):
+            # if not then create the key in container and data
+            data[k] = 0
+
+    container = {key: data[key] for key in data}
 
     for key, value in container.items():
         try:
             container[key] = keys[key]
         except KeyError:
             pass
 
-    node['avalon'].setValue('')
-    node['avalon'].setValue(toml.dumps(container))
+    node = lib.set_avalon_knob_data(node, container)
 
     return node
 
 
 class Creator(api.Creator):
+    """Creator class wrapper
+    """
     def process(self):
-        nodes = nuke.allNodes()
-
         if (self.options or {}).get("useSelection"):
             nodes = nuke.selectedNodes()
 
-        instance = [n for n in nodes
-                    if n["name"].value() in self.name] or None
+        if len(nodes) > 0:
+            node = nodes[0]
+        elif len(nodes) > 1:
+            nuke.message("Please select only one node")
```

Comment:
Doesn't this message still allow the code to continue? Or is it like raising an error? If this still continues then it seems odd. Any reason it's done like this?

Reply:
well, this all is fundamentally wrong. so I will rewrite it in the Maya's fashion.

Reply:
Default `avalon.nuke.pipeline.Creator` will now create Backdrop node with imprinted publishable metadata